plenum/server/batch_handlers/node_reg_handler.py

```diff
@@ -50,22 +52,40 @@ def post_batch_applied(self, three_pc_batch: ThreePcBatch, prev_handler_result=N
 
         three_pc_batch.node_reg = list(self.uncommitted_node_reg)
 
+        if view_no > self._uncommitted_view_no:
+            self.active_node_reg = list(self.uncommitted_node_reg)
+            self._uncommitted_view_no = view_no
+
         logger.debug("Applied uncommitted node registry: {}".format(self.uncommitted_node_reg))
         logger.debug(
             "Current node registry for previous views: {}".format(sorted(self.node_reg_at_beginning_of_view.items())))
+        logger.debug("Current active node registry: {}".format(self.active_node_reg))
 
     def post_batch_rejected(self, ledger_id, prev_handler_result=None):
         reverted = self._uncommitted.pop()
         if len(self._uncommitted) == 0:
             self.uncommitted_node_reg = self.committed_node_reg
+            self._uncommitted_view_no = self._committed_view_no
         else:
             last_uncommitted = self._uncommitted[-1]
             self.uncommitted_node_reg = last_uncommitted.uncommitted_node_reg
+            self._uncommitted_view_no = last_uncommitted.view_no
+
+        # find the uncommitted node reg at the beginning of view
+        if self._uncommitted_view_no < reverted.view_no:
+            if self._committed_view_no == self._uncommitted_view_no:
+                self.active_node_reg = list(self.node_reg_at_beginning_of_view[self._committed_view_no])
+            else:
+                i = 1
+                while i <= len(self._uncommitted) and self._uncommitted_view_no == self._uncommitted[-i].view_no:
+                    i += 1
+                self.active_node_reg = list(self._uncommitted[-i + 1].uncommitted_node_reg)
```

Comment:
Extracted into a separate method.
I believe there are already unit tests exist that cover this functionality (`test_update_node_regs_on_revert_in_different_views` and `test_update_node_regs_on_revert_in_different_views_original_view_no`). Creating new tests will be a bit redundant from my point of view, and it will have to use internal API a lot.